test/suites/pki.sh

```diff
@@ -200,9 +200,8 @@ test_pki() {
     # Remove remote
     lxc_remote remote remove pki-lxd
 
-    # Add the remote again using an incorrect token.
-    # This should succeed as is the same as the test above but with an incorrect token rather than no token.
-    lxc_remote remote add pki-lxd "${LXD5_ADDR}" --token=bar
+    # Ensure remote can be added without the token, as we are already trusted by the remote.
+    lxc_remote remote add pki-lxd "${LXD5_ADDR}" --accept-certificate
```

Comment:
This comment is a little confusing. "trusted by the remote" implies that we're in the truststore of the remote, but we aren't. We're trusted by virtue of having our certificate signed by the CA and `core.trust_ca_certificates` being enabled.

Instead of re-testing this, I think a better test would be to ensure `--accept-certificate` is not needed if the CA certificate is in the client's config directory.

Reply:
Since the same test is a few lines above I've just removed it.

The error when adding a remote, it fails because the certificate is signed by an unknown authority. Therefore, `lxc` falls back to asking whether the certificate should be accepted. Flag `--accept-certificate` papers over the issue by automatically accepting it. (`DEBUG` part is printed when CA authentication fails)
```
++ timeout --foreground 120 /home/runner/go/bin/lxc remote add pki-lxd 127.0.0.1:47377 --verbose
time="2024-10-11T11:43:10Z" level=info msg="http: TLS handshake error from 127.0.0.1:43634: EOF"

(DEBUG) Error: Get "https://127.0.0.1:47377/1.0": tls: failed to verify certificate: x509: certificate signed by unknown authority

Certificate fingerprint: 4cb5d7cebdc68022bcc864878d895637d9c1d8c55a3e1a69cc828754229b50b3
ok (y/n/[fingerprint])? 
```

I've tested this before the changes in this PR and the result is exactly the same indicating this is not a regression. 

If you agree, can this be addressed in a separate PR?

Reply:
Did you add the CA certificate to the clients configuration directory as `client.ca`? If so and it still prompts to accept the certificate then that's a bug. Happy for it to be addressed separately.

I vaguely remember that this is difficult to test - something to do with hostnames not matching perhaps.... In any case, we can address in a separate issue and if we can't test it we should add a comment to explain why for next time.